### PR TITLE
feat: complete re-design, responsive UI & dark mode

### LINKED ISSUE #12: Add light/dark mode toggle

### LINKED ISSUE #14: Mobile: make UI responsive

### PR DESCRIPTION
**Pull request by fmaclen**
Closes #12 
Closes #14 

These designs are still a moving target but I've more or less settled with this layout:

v3 - Session - light
v3 - Session -dark
v3 - New session-1

new session mobile
no sessions mobile
session mobile

### CODE REVIEW

**Comment by raychz** [comment]
I really like that you're adding in a toggle for "Auto Scroll"!  The current version of Hollama's auto scroll is problematic if you try scrolling up while the LLM is still generating text.  It also looks like you've added support for multiple knowledges/contexts (based on the screenshots above), so maybe #73 is unnecessary?

**Comment by fmaclen** [comment]
@raychz thanks!

The screenshots I included include a bunch of long-term plans, I'm not planning on this PR implementing a big chunk of it.

Here's a demo of the PR so far if you want to test-drive it: https://ui-redesign.hollama.pages.dev/settings
Colors are not final and it's only partially responsive.

> I really like that you're adding in a toggle for "Auto Scroll"! The current version of Hollama's auto scroll is problematic if you try scrolling up while the LLM is still generating text.

Yeah, it's a bit broken. I essentially want it to be the behavior to be more predictable, more or less like Claude/ChatGPT #13 

> It also looks like you've added support for multiple knowledges/contexts (based on the screenshots above), so maybe https://github.com/fmaclen/hollama/issues/73 is unnecessary?

No, it is necessary! 💯 
I won't be implementing that feature in this PR. I haven't had time to go through your suggestion in detail yet but I originally envisioned Knowledge as a collection of "knowledges" rather than a single document.

> On another note, it'd be great to add in a "Moon" icon button to the menu that will toggle dark/light mode.

Yup, you can see that in the demo linked above. I added it on the sidebar for now but I think it should look different than the other buttons. I'll likely add a separate issue for that.

File: src/routes/+layout.svelte [hunk 1]
```diff
@@ -1,13 +1,31 @@
...
+		theme = theme === 'light' ? 'dark' : 'light';
+		document.documentElement.setAttribute('data-color-theme', theme);
+	}
+
+	onMount(() => {
+		// Set initial theme based on user's browser preference
+		const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches;
+		theme = prefersDark ? 'dark' : 'light';
+		document.documentElement.setAttribute('data-color-theme', theme);
+	});
```
Comment (fmaclen) [inline src/routes/+layout.svelte:28]: REF #78

File: src/routes/+page.svelte [hunk 2]
```diff
@@ -1,186 +1,17 @@
...
+	import { beforeUpdate } from 'svelte';
+
+	// FIXME: This behavior may belong in hooks.client.ts
+	beforeUpdate(() => {
+		if (!$settingsStore?.ollamaServer) {
+			// FIXME:
+			// This should only rediect to /settings if the server is not connected
+			goto('/settings');
+		} else {
+			goto('/sessions');
```
Comment (fmaclen) [inline src/routes/+page.svelte:13]: REF #79

Comment (fmaclen) [reply src/routes/+page.svelte:13]: REF #80

**Pull request merged**